src/bots/slack/adapter.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  SlackAdapter,
+  SlackEventMiddleware,
+  SlackMessageTypeMiddleware,
+} from 'botbuilder-adapter-slack';
+
+let tokenCache: any = {};
+let userCache: any = {};
+
+const getTokenForTeam = async (teamId: string): Promise<any> => {
+  if (tokenCache[teamId]) {
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(tokenCache[teamId]);
+      }, 150);
+    });
+  } else {
+    console.error('Team not found in tokenCache: ', teamId);
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(null);
+      }, 150);
+    });
+  }
+};
+
+const getBotUserByTeam = async (teamId: string): Promise<any> => {
+  if (userCache[teamId]) {
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(userCache[teamId]);
+      }, 150);
+    });
+  } else {
+    console.error('Team not found in userCache: ', teamId);
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(null);
+      }, 150);
+    });
+  }
+};
+
+export const configSlackBotAdapter = async () => {
+  // A BotKit configuration
+  const slackAdapter: SlackAdapter = new SlackAdapter({
+    // auth token for a single-team app
+    botToken: process.env.SLACK_BOT_TOKEN,
+
+    // parameters used to secure webhook endpoint
+    verificationToken: process.env.SLACK_BOT_VERIFICATION_TOKEN,
+    clientSigningSecret: process.env.SLACK_CLIENT_SIGNING_SECRET,
+
+    // credentials used to set up oauth for multi-team apps
+    clientId: process.env.SLACK_CLIENT_ID,
+    clientSecret: process.env.SLACK_CLIENT_SECRET,
+    scopes: ['bot'],
```

Comment:
For the install endpoint to work. we need to change the scopes. The bot scope is a legacy one. 

I added a bunch of possible scopes here: https://github.com/TrimAgency/sprint-bot/pull/1/files#diff-461a9672fe884beb7c4d6097795f4cbf

I also recommend we use `oauthVersion: 'v2'`